From 62f0293ae05056fdf104f4d43b99e3532ea5be2f Mon Sep 17 00:00:00 2001
From: Stefan Hall <stefanhall@google.com>
Date: Thu, 14 May 2026 14:17:36 -0700
Subject: [PATCH] Support location expansion in build_args

---
 cocotb/cocotb.bzl | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/cocotb/cocotb.bzl b/cocotb/cocotb.bzl
index 8ab664d..461123c 100644
--- a/cocotb/cocotb.bzl
+++ b/cocotb/cocotb.bzl
@@ -94,7 +94,8 @@ def _collect_transitive_files(ctx):
 def _collect_transitive_runfiles(ctx):
     return ctx.runfiles().merge_all(
         [dep.default_runfiles for dep in ctx.attr.deps] +
-        [dep.default_runfiles for dep in ctx.attr.sim],
+        [dep.default_runfiles for dep in ctx.attr.sim] +
+        [dep.default_runfiles for dep in ctx.attr.data],
     )
 
 # Helpers for preparing test script and its environment
@@ -119,7 +120,10 @@ def _get_test_command(ctx, verilog_files, vhdl_files):
 
     includes_args = _list_to_argstring(ctx.attr.includes, "includes")
     testcase_args = _list_to_argstring(ctx.attr.testcase, "testcase")
-    build_args = _list_to_argstring(ctx.attr.build_args, "build_args")
+    # Expand $(location ...) in build_args using deps and data.
+    location_targets = ctx.attr.deps + ctx.attr.data
+    expanded_build_args = [ctx.expand_location(arg, location_targets) for arg in ctx.attr.build_args]
+    build_args = _list_to_argstring(expanded_build_args, "build_args")
     gpi_interfaces_args = _list_to_argstring(ctx.attr.gpi_interfaces, "gpi_interfaces")
     test_args = _list_to_argstring(ctx.attr.test_args, "test_args")
     plus_args = _list_to_argstring(ctx.attr.plus_args, "plus_args")
@@ -207,6 +211,11 @@ _cocotb_test_attrs = {
         doc = "Defines to set",
         default = {},
     ),
+    "data": attr.label_list(
+        doc = "Data dependencies for runtime and $(location) expansion in build_args.",
+        allow_files = True,
+        default = [],
+    ),
     "deps": attr.label_list(
         doc = "The list of python libraries to be linked in to the simulation target",
         providers = [PyInfo],
-- 
2.54.0.563.g4f69b47b94-goog

