From 90471de469c47533880628af6a5a5c034c3beb45 Mon Sep 17 00:00:00 2001
From: Stefan Hall <stefanhall@google.com>
Date: Wed, 20 May 2026 15:33:08 -0700
Subject: [PATCH] Fix runfiles collection for direct files in data attribute

---
 cocotb/cocotb.bzl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cocotb/cocotb.bzl b/cocotb/cocotb.bzl
index 70a6d21..74be598 100644
--- a/cocotb/cocotb.bzl
+++ b/cocotb/cocotb.bzl
@@ -96,7 +96,7 @@ def _collect_transitive_files(ctx):
     )
 
 def _collect_transitive_runfiles(ctx):
-    return ctx.runfiles().merge_all(
+    return ctx.runfiles(files = ctx.files.data).merge_all(
         [dep.default_runfiles for dep in ctx.attr.deps] +
         [dep.default_runfiles for dep in ctx.attr.sim] +
         [dep.default_runfiles for dep in ctx.attr.data],
-- 
2.45.0.118.g7fe29c98d7-goog

