From 75fbfd6f32c8a72d3a96ec290dbe64c2754b0229 Mon Sep 17 00:00:00 2001
From: Alex Van Damme <atv@google.com>
Date: Tue, 5 May 2026 11:47:33 -0700
Subject: [PATCH 11/11] Use short_path for python runfiles resolution

---
 cocotb/cocotb.bzl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cocotb/cocotb.bzl b/cocotb/cocotb.bzl
index 90c70bc..afba893 100644
--- a/cocotb/cocotb.bzl
+++ b/cocotb/cocotb.bzl
@@ -136,7 +136,7 @@ def _get_test_command(ctx, verilog_files, vhdl_files):
     seed_args = " --seed {}".format(ctx.attr.seed) if ctx.attr.seed != "" else ""
 
     test_module_args = _pymodules_to_argstring(ctx.files.test_module, "test_module")
-    python_interpreter = ctx.toolchains["@rules_python//python:toolchain_type"].py3_runtime.interpreter.path
+    python_interpreter = ctx.toolchains["@rules_python//python:toolchain_type"].py3_runtime.interpreter.short_path
 
     command = (
         "PATH={}:$PATH ".format(_get_path_to_set(ctx)) +
-- 
2.54.0.563.g4f69b47b94-goog

