diff -ur cvfpu-58ca3c376beb914b2b80b811d4b270c063d4e6f7/src/fpnew_divsqrt_multi.sv cvfpu-new2/src/fpnew_divsqrt_multi.sv
--- cvfpu-58ca3c376beb914b2b80b811d4b270c063d4e6f7/src/fpnew_divsqrt_multi.sv	2025-10-20 22:42:20.000000000 -0700
+++ cvfpu-new2/src/fpnew_divsqrt_multi.sv	2026-03-18 17:07:47.782731037 -0700
@@ -184,7 +184,7 @@
   logic unit_busy;              // valid data in flight
   logic simd_synch_done;
   // FSM states
-  typedef enum logic [1:0] {IDLE, BUSY, HOLD} fsm_state_e;
+  typedef enum logic [1:0] {IDLE, BUSY, HOLD, UNREACHABLE} fsm_state_e;
   fsm_state_e state_q, state_d;
 
   // Valids are gated by the FSM ready. Invalid input ops run a sqrt to not lose illegal instr.
diff -ur cvfpu-58ca3c376beb914b2b80b811d4b270c063d4e6f7/src/fpnew_divsqrt_th_64_multi.sv cvfpu-new2/src/fpnew_divsqrt_th_64_multi.sv
--- cvfpu-58ca3c376beb914b2b80b811d4b270c063d4e6f7/src/fpnew_divsqrt_th_64_multi.sv	2025-10-20 22:42:20.000000000 -0700
+++ cvfpu-new2/src/fpnew_divsqrt_th_64_multi.sv	2026-03-18 17:07:47.786731034 -0700
@@ -205,7 +205,7 @@
   logic unit_busy;              // valid data in flight
   logic simd_synch_done;
   // FSM states
-  typedef enum logic [1:0] {IDLE, BUSY, HOLD} fsm_state_e;
+  typedef enum logic [1:0] {IDLE, BUSY, HOLD, UNREACHABLE} fsm_state_e;
   fsm_state_e state_q, state_d;
 
   // Valids are gated by the FSM ready. Invalid input ops run a sqrt to not lose illegal instr.
diff -ur cvfpu-58ca3c376beb914b2b80b811d4b270c063d4e6f7/src/fpnew_pkg.sv cvfpu-new2/src/fpnew_pkg.sv
--- cvfpu-58ca3c376beb914b2b80b811d4b270c063d4e6f7/src/fpnew_pkg.sv	2025-10-20 22:42:20.000000000 -0700
+++ cvfpu-new2/src/fpnew_pkg.sv	2026-03-18 17:07:47.776731042 -0700
@@ -144,6 +144,7 @@
     RUP = 3'b011,
     RMM = 3'b100,
     ROD = 3'b101,  // This mode is not defined in RISC-V FP-SPEC
+    R_UNREACH_1 = 3'b110,
     DYN = 3'b111
   } roundmode_e;
 
